elide-datastore/elide-datastore-aggregation/src/main/java/com/yahoo/elide/datastores/aggregation/queryengines/sql/SQLQueryEngine.java

```diff
@@ -129,127 +144,187 @@ public MetricProjection constructMetricProjection(Metric metric,
+        resultBuilder.data(new SQLEntityHydrator(resultSet, query, getMetadataDictionary()).hydrate());
         return resultBuilder.build();
     }
 
-    private long getPageTotal(Query query, SQLQuery sql, EntityManager entityManager) {
-        String paginationSQL = toPageTotalSQL(sql, dialect).toString();
-
-        javax.persistence.Query pageTotalQuery =
-                entityManager.createNativeQuery(paginationSQL)
-                        .setHint(QueryHints.HINT_READONLY, true);
+    private long getPageTotal(Query query, SQLQuery sql, SqlTransaction sqlTransaction) {
+        String paginationSQL = toPageTotalSQL(sql, sqlTransaction.dialect).toString();
 
-        //Supply the query parameters to the query
-        supplyFilterQueryParameters(query, pageTotalQuery);
+        NamedParamPreparedStatement stmt = sqlTransaction.initializeStatement(paginationSQL);
 
-        //Run the Pagination query and log the time spent.
-        return new TimedFunction<>(
-                () -> CoerceUtil.coerce(pageTotalQuery.getSingleResult(), Long.class),
-                "Running Query: " + paginationSQL
+        // Supply the query parameters to the query
+        supplyFilterQueryParameters(query, stmt);
+
+        // Run the Pagination query and log the time spent.
+        return new TimedFunction<>(() -> {
+            try {
+                ResultSet rs = stmt.executeQuery();
+                if (rs.next()) {
+                    return rs.getLong(1);
+                } else {
+                    return null;
+                }
+            } catch (SQLException e) {
+                throw new IllegalStateException(e);
+            }
+        }, "Running Query: " + paginationSQL
         ).get();
     }
 
     @Override
     public String getTableVersion(Table table, Transaction transaction) {
-        EntityManager entityManager = ((SqlTransaction) transaction).entityManager;
+        SqlTransaction sqlTransaction = (SqlTransaction) transaction;
 
         String tableVersion = null;
         Class<?> tableClass = getMetadataDictionary().getEntityClass(table.getName(), table.getVersion());
         VersionQuery versionAnnotation = tableClass.getAnnotation(VersionQuery.class);
         if (versionAnnotation != null) {
             String versionQueryString = versionAnnotation.sql();
-            javax.persistence.Query versionQuery =
-                    entityManager.createNativeQuery(versionQueryString)
-                            .setHint(QueryHints.HINT_READONLY, true);
-            tableVersion = new TimedFunction<>(
-                    () -> CoerceUtil.coerce(versionQuery.getSingleResult(), String.class),
-                    "Running Query: " + versionQueryString
+            NamedParamPreparedStatement stmt = sqlTransaction.initializeStatement(versionQueryString);
+            tableVersion = new TimedFunction<>(() -> {
```

Comment:
You could commonize this code with a template function that takes another function with a result set argument

```java
<T> runQuery(String query, Function<ResultSet, T> resultSetMapper)
```
or something along those lines.

Reply:
updated.